src/schoology-extractor/tests/helpers/test_export_data.py

```diff
@@ -29,5 +30,18 @@ def test_then_call_DataFrame_method(self, mocker):
             # Assert
             DataFrame_mock.assert_called_once()
 
+    class Test_when_to_string_method_is_called():
+        def test_given_no_data_parameter_then_throw_assert_exception(self):
+            with pytest.raises(AssertionError):
+                to_string(None)   # type: ignore
 
+        def test_then_call_DataFrame_method(self, mocker):
 
+            # Arrange
+            DataFrame_mock = mocker.patch.object(pd, "DataFrame")
+
+            # Act
+            to_string([])
+
+            # Assert
+            DataFrame_mock.assert_called_once()
```

Comment:
This is an example of over-specification in a test, which is one of the dangers of mocking in general. It would be preferable to define the expected output and not mock the DataFrame. Just confirm that the actual output matches the expected output. Then if someone wants to change the code to get the expected output without pandas, they don't need to change the unit test as well.

Reply:
✅

Reply:
Changed ✅